sample-service/src/main/java/uk/ac/ox/ndph/mts/sample_service/security/authorisation/SiteTreeUtil.java

```diff
@@ -0,0 +1,61 @@
+package uk.ac.ox.ndph.mts.sample_service.security.authorisation;
+
+import com.google.common.collect.Lists;
+import org.springframework.stereotype.Component;
+import uk.ac.ox.ndph.mts.sample_service.client.dtos.SiteDTO;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class SiteTreeUtil {
```

Comment:
Going on past experience this method - get accessible subtrees of the sites tree - could be used a lot - might be worth having an endpoint in site-service to return it (filtering using these methods). Not necessary right now.